python/examples/minimal_build/build_venv.sh

```diff
@@ -68,11 +69,9 @@ export CMAKE_PREFIX_PATH=${ARROW_HOME}${CMAKE_PREFIX_PATH:+:${CMAKE_PREFIX_PATH}
 export PYARROW_BUILD_TYPE=Debug
 export PYARROW_CMAKE_GENERATOR=Ninja
 
-# You can run either "develop" or "build_ext --inplace". Your pick
-
-# python setup.py build_ext --inplace
-python setup.py develop
+# Use the same command that we use on python_build.sh
+python -m pip install --no-deps --no-build-isolation -e .
```

Comment:
Do you know why the `-e` is needed if then below we actually run tests against installed package? (now, we did install it in editable mode before as well, so fine to just keep this, just wondering why the change to `--pyargs pyarrow` below if we keep this editable)

Reply:
Maybe also keep the `-v` or `-vv` you had in a previous commit, will be useful for debugging CI issues I think

Reply:
I am not sure why the editable mode is required but otherwise it fails importing `pyarrow.lib`. I have been able to validate on my local build tree that without the editable mode the cython compiled files aren't available there, I should investigate further. Those files aren't there without it, I should investigate where are those being installed because they are being generated:
```
python/pyarrow/lib.cpython-310-x86_64-linux-gnu.so
python/pyarrow/lib.cpython-38-x86_64-linux-gnu.so
```
as per the `--pyargs pyarrow` I just copied the same command we use for testing on `python_test.sh` when I was debugging, I can probably revert back this change.

Reply:
> I am not sure why the editable mode is required but otherwise it fails importing `pyarrow.lib`.

That's probably because the current directory contains the `pyarrow` source directory. If you change the current directory to something else, the error should probably disappear.

Reply:
thanks @pitrou this was a good pointer! I've pushed a change https://github.com/apache/arrow/pull/41041/commits/410f0e0539346b4d5a3aa6b7e284482042e992e8 that removes the editable install requirement